rednotebook/external/text2rednotebook.py

```diff
@@ -0,0 +1,145 @@
+"""
+Mass import diary/journal entries from markdown/plain text/odt into rednotebook
+"""
+
+import argparse
+import sys
+import re
+import datetime
+from odf import text
+from odf.opendocument import load
+sys.path.append('/usr/share/rednotebook')
+
+# must import rednotebook after munging path in order to find it, as it is
+# not (at least in Debian) installed in the default python path
+import rednotebook.storage  # pylint: disable=wrong-import-position, unused-import
+
+def main():
+    """ parse commandline arguments & process text """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-n', '--dry-run', action='store_true')
+    parser.add_argument('--echo-entries', action='store_true')
+    parser.add_argument('--existing', choices=['overwrite', 'append', 'skip', 'error'],
+                        default='error')
+    parser.add_argument('--no-list-missing-entries', action='store_true')
+    parser.add_argument('infile')
+    args = parser.parse_args()
+    if args.infile.endswith('.odt'):
+        textdoc = load(args.infile)
+        alltext = [str(para) for para in textdoc.getElementsByType(text.P)]
+        alltext = "\n\n".join(alltext)
+    else:
+        with open(args.infile) as fin:
+            alltext = fin.read()
+
+    # remove non-printable characters
+    control_chars = ''.join(map(chr, range(0x7f,0xa0)))
+    control_char_re = re.compile('[%s]' % re.escape(control_chars))
+    alltext = control_char_re.sub('', alltext)
+
+    days = re.split(r'(\w+day[,]?\s*\d+\s+\w+\s+\d+)\.?\s*', alltext, flags=re.IGNORECASE)
+    i = 0
+    if days:
+        print(f'Found entries for {(len(days)-1)//2} days')
+        if days[0] != '':
+            print(f"Ignoring leading text: {days[0]}")
+        i += 1
+
+    # have to import *after* munging argv as rednotebook does its own argv parsing
+    sys.argv = [sys.argv[0]]
+    import rednotebook.journal  # pylint: disable=import-outside-toplevel, redefined-outer-name
+    journal = rednotebook.journal.Journal()
+    data_dir = journal.get_journal_path()
```

Comment:
I would make the data_dir a parameter of the script. Otherwise, it's too easy to overwrite data in an existing journal.

Reply:
Done